### PR TITLE
Add workshop setup instructions to the website

### PR DESCRIPTION
**Pull request by GilHenriques**
- Added a new menu category: *Installation Instructions > Workshop Setup*, which in turn has pages for the Intro R, Tidyverse, and Reproducibility workshops.
- These pages provide setup instructions for each workshop, with links to pre-existing installation instruction pages where appropriate
- The pages also include download buttons for the data sets using during the workshops (avoiding the use of weblinks for importing data during workshops)
- One potential issue (which may or may not be a problem) is that the buttons that download the data files get those data from the EDUCE-UBC/workshop_data folder (via their respective weblinks); i.e. the data files are not stored internally in the website repository. I chose to do this to avoid having duplicate data files in different repositories. Not sure if that was the right call.
- It may be appropriate at some point to create an Installation Instruction page for Git and GitHub (which make up the core of the Reproducibility setup page). If we decide to do so we need to figure out what to do re: the Windows command line. Currently the Reproducibility setup page advises the use of GitBash because it installs Git in a way that can be directly configured in RStudio. It is unclear to me whether the same is true for the WSL. Something to look into for the future...

### CODE REVIEW

File: r_and_rstudio_ws_setup.Rmd [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: "Workshop setup instructions"
+subtitle: "Workshop: Introduction to R and RStudio"
+output: html_document
+params:
+  r_version: 4.0.2
+  r_legacy_version: 3.3.3
+  rstudio_version: 1.3.1073
+  rstudio_legacy_version: 1.1.463
```
Comment (stephan-koenig) [inline r_and_rstudio_ws_setup.Rmd]:
```suggestion
```
The `params` are only needed in `r_and_rstudio.Rmd`.

File: reproducibility_ws_setup.Rmd [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+title: "Workshop setup instructions"
+subtitle: "Workshop: Reproducible research"
+output: 
+  html_document:
+    toc: yes
+params:
+  r_version: 4.0.2
+  r_legacy_version: 3.3.3
+  rstudio_version: 1.3.1073
+  rstudio_legacy_version: 1.1.463
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
params:
  make_version: 4.3
```
`params` for R and RStudio are not needed. Use params for the latest version of `make` and use it below.

File: reproducibility_ws_setup.Rmd [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    class = "hvr-sweep-to-left",
+    csv2 = FALSE
+  )
+```
+
+## Instructions for Windows users
+
+### Terminal
+
+Install [GitBash](https://gitforwindows.org/), which emulates Bash (the Unix shell) in Windows. All command line instructions should be completed in GitBash and not the native Windows Command Prompt.
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
Install [GitBash](https://gitforwindows.org/), which emulates Bash (a Unix shell) in Windows. All command-line instructions should be completed in GitBash and not the native Windows Command Prompt.
```

File: reproducibility_ws_setup.Rmd [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+![](images/setup_git_windows.png)
+
+### GitHub
+
+1. If you do not already have one, sign up for an account on [GitHub](https://github.com/).
+2. Configure your GitHub user name and email (the email address you used to register on GitHub). To do this, open GitBash and type:
+
+```
+git config --global user.name 'yourGitHubUsername'
+git config --global user.email 'name@provider.com'
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
git config --global user.name '<your GitHub username>'
git config --global user.email '<name@provider.com>'
```

File: reproducibility_ws_setup.Rmd [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+1. Download the [latest version of Make](https://sourceforge.net/projects/ezwinports/files/) *without guile*.
+  * The file name looks like `make-4.1-2-without-guile-w32-bin.zip`, though the most recent version number may be different.
+2. Unzip the folder.
+3. Copy all of its contents to `c/Program Files/Git/mingw64/`
+  * Merge all of the contents of the folders but do NOT overwrite/replace any existing files.
+4. Check that Make is installed by typing `make --version` [Enter] in GitBash. This should populate the window with information on your version of Make such as
+
+```
+GNU Make 4.2.1
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
GNU Make 4.3
```

File: reproducibility_ws_setup.Rmd [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+```
+git config --global user.name 'yourGitHubUsername'
+git config --global user.email 'name@provider.com'
+```
+
+### Make
+
+1. Download the [latest version of Make](https://sourceforge.net/projects/ezwinports/files/) *without guile*.
+  * The file name looks like `make-4.1-2-without-guile-w32-bin.zip`, though the most recent version number may be different.
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
1. Download the [Make *without guile* (version `r params$make_version)`](https://sourceforge.net/projects/ezwinports/files/), file name `` `r paste0("make-", params$make-version, "-without-guile-w32-bin.zip")` ``
```
Here an example of how to use a params variable.

File: reproducibility_ws_setup.Rmd [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+2. Unzip the folder.
+3. Copy all of its contents to `c/Program Files/Git/mingw64/`
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
3. Copy all of its contents to `C:\Program Files\Git\mingw64\`.
```

I think it is better to stick with Windows formatting of file paths.

File: reproducibility_ws_setup.Rmd [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+License GPLv3+: GNU GPL verison 3 or later <http://gnu.org/licenses/gpl.html>
+This is free software: you are free to change and redistribute it.
+There is NO WARRANTY, to the extent permitted by law.
+```
+
+## Instructions for Mac users
+
+### Terminal
+
+The native Terminal in the Mac OS is Linux-based and therefore, can be used in this workshop. You can access it under Applications > Utilities > Terminal.app
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
The native Terminal on macOS is Unix-based and, therefore, can be used in this workshop. You can access it under Applications > Utilities > Terminal.app
```

File: reproducibility_ws_setup.Rmd [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+```
+git version 2.18.0
+```
+
+3. Once you've installed Git, you'll need to activate it in RStudio:
+  * Open RStudio and go to Tools > Global Options > Git/SVN
+  * Click "Enable version control interface for RStudio projects"
+  * If necessary, enter the path for your Git executable where provided (see picture below).
+  * Restart RStudio, if it worked out you will find the Git icon on the top toolbar, as shown below.
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]: Use `-` for bullet points.

File: reproducibility_ws_setup.Rmd [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+2. Configure your GitHub user name and email (the email address you used to register on GitHub). To do this, open the Terminal and type:
+
+```
+git config --global user.name 'yourGitHubUsername'
+git config --global user.email 'name@provider.com'
+```
+
+### Make
+
+1. Download and install the Mac command line developer tools (including Make) by opening your Terminal and typing: `xcode-select --install` [Enter]. Click "Install" to allow the installation of the full developer’s packet.
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]: You should clarify that a window opens (outside of the terminal).

File: reproducibility_ws_setup.Rmd [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+![](images/setup_git_mac.png)
+
+### GitHub
+
+1. If you do not already have one, sign up for an account on [GitHub](https://github.com/).
+2. Configure your GitHub user name and email (the email address you used to register on GitHub). To do this, open the Terminal and type:
+
+```
+git config --global user.name 'yourGitHubUsername'
+git config --global user.email 'name@provider.com'
```
Comment (stephan-koenig) [inline reproducibility_ws_setup.Rmd]:
```suggestion
git config --global user.name '<your GitHub username>'
git config --global user.email '<name@provider.com>'
```

File: tidyverse_ws_setup.Rmd [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+output: html_document
+params:
+  r_version: 4.0.2
+  r_legacy_version: 3.3.3
+  rstudio_version: 1.3.1073
+  rstudio_legacy_version: 1.1.463
+---
+
+- Make sure that you have installed the [latest versions of R and RStudio](r_and_rstudio.html). 
+- Install the tidyverse package by copying the following code into the Console, after the prompt `>`
```
Comment (stephan-koenig) [inline tidyverse_ws_setup.Rmd]:
```suggestion
- In RStudio, install the tidyverse package by copying the following code into the Console, after the prompt `>`.
```
I would direct them to install the educer package, which should then install all of the necessary packages for the tidyverse. At least, I would add educer as an extra step.

File: tidyverse_ws_setup.Rmd [hunk 13]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Workshop setup instructions"
+subtitle: "Workshop: Introduction to the R tidyverse"
+output: html_document
+params:
+  r_version: 4.0.2
+  r_legacy_version: 3.3.3
+  rstudio_version: 1.3.1073
+  rstudio_legacy_version: 1.1.463
```
Comment (stephan-koenig) [inline tidyverse_ws_setup.Rmd]:
```suggestion
params:
  r_version: 4.0.2
  r_legacy_version: 3.3.3
  rstudio_version: 1.3.1073
  rstudio_legacy_version: 1.1.463
```

**Review by stephan-koenig — DISMISSED** [review]
Looks good! I feel there are quite some redundancies. Maybe we can solve them at some point by using child documents.

File: reproducibility_ws_setup.Rmd (hunk 5 above)
Comment (GilHenriques) [reply reproducibility_ws_setup.Rmd]: Is there any way to use `params$make_version` within a code chunk delimited by ```, such as this one?

**Comment by GilHenriques** [comment]
Hi,
I addressed all of the changes you suggested (thank you for those!). A review still seems to be required.

File: reproducibility_ws_setup.Rmd (hunk 5 above)
Comment (stephan-koenig) [reply reproducibility_ws_setup.Rmd]: I believe you can just make another chunk with either one or two backticks `` ` ``.

Comment (stephan-koenig) [reply reproducibility_ws_setup.Rmd]: e.g. `` `r params$make_version` ``.

Comment (GilHenriques) [reply reproducibility_ws_setup.Rmd]: Thanks! I fixed it now, and also fixed another `params`-related error that was stopping the file from building (it builds fine now).

**Comment by stephan-koenig** [comment]
Can you re-request the code review (see image on Slack conversation)? I am curious if that's the way it works.

**Review by stephan-koenig — APPROVED** [review]
Great!

**Pull request merged**